x-pack/solutions/security/plugins/security_solution/public/management/pages/event_filters/view/components/event_filters_flyout.tsx

```diff
@@ -88,16 +89,23 @@ export const EventFiltersFlyout: React.FC<EventFiltersFlyoutProps> = memo(
             },
           })
         );
-        setEnrichedData({
+        const enriched = {
           ...data,
           host: {
             ...data.host,
             os: {
               ...(data?.host?.os || {}),
-              name: [searchResponse.rawResponse.hits.hits[0]._source.host.os.name],
+              name: [searchResponse.rawResponse.hits.hits[0]._source.host.os.type],
```

Comment:
```suggestion
              type: [searchResponse.rawResponse.hits.hits[0]._source.host.os.type],
```

and also in `osTypeBasedOnAgentType` below update the logic to use `type` instead.

Reply:
Thanks for the approval and the suggestion! I appreciate the input, but I'd prefer to keep this PR focused on the bug fix rather than refactoring the enrichment logic at this point.